### PR TITLE
Decouple dependency check from updating mods

### LINKED ISSUE #1745: [8.0-develop]: Dependency check doesn't work between CurseForge and Modrinth
### Operating System

Windows

### Version of Prism Launcher

8.0-develop

### Version of Qt

Qt 6.6.0

### Description of bug

Build used: https://github.com/PrismLauncher/PrismLauncher/actions/runs/6605886969

If you have a mod installed from Modrinth which has a required dependency
And you have said required dependency installed but from CurseForge (or the other way around)
Then it won't be able to find it and therefor try to download / recommend it on every update check

This is also relevant for modified versions of a mod (see Just Enough Resources in this screenshot)
It doesn't recognize it as the dependency it needs

image

image

image

### Steps to reproduce

See description

### Suspected cause

_No response_

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### LINKED ISSUE #1814: Rename "Delete metadata" to convey its goal
### Role

I play modded Minecraft

### Suggestion

I suggest changing "delete metadata" to "reset download location"

### Benefit

Because it sounds less destructive and more obvious

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

In https://github.com/PrismLauncher/PrismLauncher/issues/1528, a "delete metadata" button was added. I wondered what it does until I read that issue.

As the issue explains, it only changes the _download location that Prism uses to update it_. It does not nullify mod's manifest, remove version restrictions or something like that. It doesn't even gray out after clicked.

Hence, I suggest making the name more obvious to the users. "Reset download location" may be a bit long, so I'm open to alternative names that would still better convey the idea.

### PR DESCRIPTION
**Pull request by Trial97**
Added a button to check for updates and missing dependencies.
And kept the original button just to update the mods.
fixes #1745
fixes #1814

### CODE REVIEW

**Review by getchoo — APPROVED** [review]

**Comment by getchoo** [comment]
i don't see any reason why this needs to wait til 8.1. works as intended in my testing

**Comment by Trial97** [comment]
I added most of my new PRs to that milestone so I do not slow down the release of 8.0

**Comment by TheKodeToad** [comment]
I don't like this implementation, because there will be too many buttons :catquake:

**Comment by Trial97** [comment]
I'm okay with removing this from milestone in favor of https://github.com/PrismLauncher/PrismLauncher/pull/1768 or https://github.com/PrismLauncher/PrismLauncher/pull/1796

**Comment by Trial97** [comment]
> I don't like this implementation, because there will be too many buttons :catquake:

Then you will love what I did  to it now (you have fewer visible buttons)

**Comment by Madis0** [comment]
Should dependencies even be conceptually linked to updates? 

As a user, I would maybe assume to find it in the mod addition window instead. For example, I pick a couple mods and there is a button to pick the mods' dependencies for me as well. Though that would not cover existing instance use cases so I'm not fully sure.

Edit: considering 8.0 already does kind of what I suggested (without showing dependencies as I pick mods), then that use case is already covered and this PR's menu can be added for existing instance use cases, indeed.

**Comment by Trial97** [comment]
It is linked to updates because you can have missing dependencies and you will want to check that (by other means than downloading all mods again through the mod download dialog).
With this PR though the default button will only update(without deps) and there will be a submenu for the dependency check.

**Comment by Desoroxxx** [comment]
> Should dependencies even be conceptually linked to updates?

Mods can add new dependencies with updates.

File: launcher/ui/pages/instance/ModFolderPage.cpp [hunk 1]
```diff
@@ -84,50 +84,69 @@ ModFolderPage::ModFolderPage(BaseInstance* inst, std::shared_ptr<ModFolderModel>
...
+            update->setToolTip(tr("Try to check or update all selected mods (all mods if none are selected)"));
+            connect(update, &QAction::triggered, this, &ModFolderPage::updateMods);
+        }
+        if (!APPLICATION->settings()->get("ModDependenciesDisabled").toBool()) {  // dependencies
+
+            auto updateWithDeps = updateMenu->addAction(tr("Verify Dependencies"));
+            updateWithDeps->setToolTip(
+                tr("Try to update and check for missing dependencies all selected mods (all mods if none are selected)"));
+            connect(updateWithDeps, &QAction::triggered, this, [this] { updateMods(true); });
+        }
```
Comment (TheKodeToad) [inline launcher/ui/pages/instance/ModFolderPage.cpp]: Maybe make it sync with the setting e.g. (not tested)

```suggestion
        auto update = updateMenu->addAction(tr("Check for Updates"));
        update->setToolTip(tr("Try to check or update all selected mods (all mods if none are selected)"));
        connect(update, &QAction::triggered, this, &ModFolderPage::updateMods);

        auto updateWithDeps = updateMenu->addAction(tr("Verify Dependencies"));
        updateWithDeps->setToolTip(
            tr("Try to update and check for missing dependencies all selected mods (all mods if none are selected)"));
        connect(updateWithDeps, &QAction::triggered, this, [this] { updateMods(true); });

        auto depsDisabled = APPLICATION->settings()->getSetting("ModDependenciesDisabled");
        updateWithDeps->setVisible(!depsDisabled->get().toBool());
        connect(depsDisabled.get(), &Setting::SettingChanged, this,
                [](const Setting& setting, QVariant value) { updateWithDeps->setVisible(!value.toBool()); });
```

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**